### PR TITLE
feat: add travis config for CI/CD and sonar project props

### PR DESCRIPTION
**Pull request by pllearns**

### CODE REVIEW

**Comment by pllearns** [comment]
@oldsquarehead Do you mind taking a look at my travis config. I think the gitmodules are defined correctly, but I can't get past that error to see if the travis config is good. Thanks

File: .gitmodules [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [submodule "Carthage/Checkouts/promises"]
-	path = SampleMacOS/Carthage/Checkouts/promises
```
Comment (oldsquarehead) [inline .gitmodules]: I think the path here is wrong, it should be
```
[submodule "Carthage/Checkouts/promises"]
        path = Carthage/Checkouts/promises
        url = https://github.com/google/promises.git
```

Comment (pllearns) [reply .gitmodules]: Thanks, I did update the SampleMacOS to `sdk-ble-swift` but I didn't push that change. I'll definitely try the path as you wrote it. Thanks @oldsquarehead

**Comment by arietrouw** [comment]
Broken?

**Comment by oldsquarehead** [comment]
FYI the issue here was 
- Podfile in the root that didn't need to be there
- Building the library vs. the sample project
Works now, and I've merged these changes into my `release/boston5.1.5` branch, so I'll close this and open a new one with the new branch. Thanks.

**Pull request closed without merging**